src/access-provider.ts

```diff
@@ -0,0 +1,65 @@
+import * as rp from 'request-promise-native';
+
+export class AccessProvider {
+  private apiAuthorization: any;
+  private apiAccessTokenPromise: Promise<any> | undefined;
+
+  constructor(public trustpilotApiConfig: Readonly<ITrustpilotApiConfig>) {
+  }
+
+  private createApiTokenRequest(): any {
+    return this.trustpilotApiConfig.apiTokenRequest || {
+      uri: '/v1/oauth/oauth-business-users-for-applications/accesstoken',
+      form: {
+        'grant_type': 'password',
+        username: this.trustpilotApiConfig.apiUsername,
+        password: this.trustpilotApiConfig.apiPassword
+      }
+    };
+  }
+
+  private async createApiAccessToken(): Promise<any> {
+    this.apiAuthorization = await rp.defaults({
+      baseUrl: this.trustpilotApiConfig.apiBaseUrl,
```

Comment:
I think we need to handle the invitations API case, as was done [in my PR](https://github.com/trustpilot/node-trustpilot/pull/43/commits/2ffb86f1c8daec83e4d2cccd53cf482e0a30ddd4#diff-d7d2e16fc7d474f4bcea971b275f0752R27).